src/main/java/org/json/JSONTokener.java

```diff
@@ -394,51 +394,158 @@ public String nextTo(String delimiters) throws JSONException {
         }
+
         return nextSimpleValue(c);
     }
 
+    private boolean checkIfValueIsNumeric(Object valueToValidate) {
+        for (char c : valueToValidate.toString().toCharArray()) {
+            if (!Character.isDigit(c)) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    /**
+     * This method is used to get a JSONObject from the JSONTokener. The strictMode parameter controls the behavior of
+     * the method when parsing the JSONObject.
+     *
+     * @param strictMode If true, the method will strictly adhere to the JSON syntax, throwing a JSONException for any
+     *                   deviations.
+     * @return A JSONObject which is the next value in the JSONTokener.
+     * @throws JSONException If the JSONObject or JSONArray depth is too large to process.
+     */
+    private JSONObject getJsonObject(boolean strictMode) {
+        try {
+            if (strictMode) {
+                return new JSONObject(this, new JSONParserConfiguration().withStrictMode(true));
+            }
+
+            return new JSONObject(this);
+        } catch (StackOverflowError e) {
+            throw new JSONException("JSON Array or Object depth too large to process.", e);
+        }
+    }
+
+    /**
+     * This method is used to get a JSONArray from the JSONTokener.
+     *
+     * @return A JSONArray which is the next value in the JSONTokener.
+     * @throws JSONException If the JSONArray depth is too large to process.
+     */
+    private JSONArray getJsonArray() {
+        try {
+            return new JSONArray(this);
+        } catch (StackOverflowError e) {
+            throw new JSONException("JSON Array or Object depth too large to process.", e);
+        }
+    }
+
+    /**
+     * This method checks if the provided value is wrapped by quotes.
+     *
+     * @param valueToValidate The value to be checked. It is converted to a string before checking.
+     * @return A boolean indicating whether the value is wrapped by quotes. It returns true if the value is wrapped by
+     * either single or double quotes.
+     */
+    private boolean valueIsWrappedByQuotes(Object valueToValidate) {
+        String stringToValidate = valueToValidate.toString();
+        boolean isWrappedByDoubleQuotes = isWrappedByQuotes(stringToValidate, "\"");
+        boolean isWrappedBySingleQuotes = isWrappedByQuotes(stringToValidate, "'");
+        return isWrappedByDoubleQuotes || isWrappedBySingleQuotes;
+    }
+
+    private boolean isWrappedByQuotes(String valueToValidate, String quoteType) {
+        return valueToValidate.startsWith(quoteType) && valueToValidate.endsWith(quoteType);
+    }
+
     Object nextSimpleValue(char c) {
-        String string;
+        return nextSimpleValue(c, false);
+    }
 
-        switch (c) {
-        case '"':
-        case '\'':
-            return this.nextString(c);
+    Object nextSimpleValue(char c, boolean strictMode) {
+        if (c == '"' || c == '\'') {
```

Comment:
@rikkarth It seems to me that strings delimited by single quotes should not be allowed in strict mode.